### DISCUSSION TITLE
How can i non-axisymmetric planewave sources be used in cylindrical coordinates?

### DISCUSSION THREAD

**Discussion by semiuse** [NanoComp/meep#2523] [Q&A]
I want to create a planewave incident at an angle of 30 degrees while using a cylindrical coordinate system.
(That is, the structure is cylindrically symmetric in the Z axis, but the source is not symmetric.)
Are there any examples related to this?

**Comment by smartalecH** [comment]
[Here's a tutorial for nonaxisymmetric dipole sources](https://meep.readthedocs.io/en/latest/Python_Tutorials/Cylindrical_Coordinates/#nonaxisymmetric-dipole-sources). You can do something similar for an arbitrary source (like a planewave source pointing in an arbitrary direction). 

In short it will require various simulations run with different m numbers and using different polarizations (to get the planewave). 

Even though you have to run several simulations, my hunch is that it's still faster than a single large 3d simulation.

**Reply by semiuse** [reply]
Thanks for your answer.

If the source is arranged with center(4,0,0), size=(8,0,0), an error occurs when m is greater than 1 because of the point where r=0. Is there any way to solve it?

**Comment by stevengj** [comment]
As mentioned in https://github.com/NanoComp/meep/issues/902, you can use a [Jacobi–Anger expansion](https://en.wikipedia.org/wiki/Jacobi%E2%80%93Anger_expansion) for off-axis planewaves.

An example can be found in equations 5–7 of [this paper](https://math.mit.edu/~stevenj/papers/LinCa21.pdf), for example.

**Reply by semiuse** [reply]
Thanks for your answer.

I will refer to the paper you mentioned.
